src/OrchardCore.Modules/OrchardCore.Media/Drivers/MediaFieldDriver.cs

```diff
@@ -110,6 +122,16 @@ public override async Task<IDisplayResult> UpdateAsync(MediaField field, IUpdate
                 // If it's an attached media field editor the files are automatically handled by _attachedMediaFieldFileService
                 if (string.Equals(context.PartFieldDefinition.Editor(), "Attached", StringComparison.OrdinalIgnoreCase))
                 {
+                    for (var i = 0; i < items.Count; i++)
+                    {
+                        if (items[i].IsNew)
+                        {
+                            var lastSlash = items[i].Path.LastIndexOf('/');
+                            var nameOnly = items[i].Path[(lastSlash + 1)..];
+                            items[i].AttachedFileName = nameOnly[36..]; // remove unique id from name
```

Comment:
It would be better if the javascript app includes the `AttachedFileName` on it's  post, than having to do this .

Reply:
I removed this code in https://github.com/OrchardCMS/OrchardCore/pull/10570/commits/99ece1690cb60ed361b68ef50f7b7924805aa985 but I'm still trying to figure out where to include it in the JavaScript app.

Reply:
@deanmarcussen, If I understand how this works correctly, when you select or drop in a file it immediately posts the file and saves it to a temporary location and then moves it once you save the content type the file is attached to.  Is that correct and if so, at which step should the AttachedFileName be set?

Reply:
I think I figured it out but I'm not sure I did it the best way.  Stepping through the JavaScript I did not see another way but I may have missed something.  It's in this commit: https://github.com/OrchardCMS/OrchardCore/pull/10570/commits/481d4f3e282148802fe8a59bb16d9d5205851121